openedx/course-discovery
# feat: transform contentful fields and index in algolia

[PROD-3024](https://2u-internal.atlassian.net/browse/PROD-3024)
This PR adds a util functions which are used to transform bootcamp data from Contentful. The transformed fields are going to be sent to algolia for indexing and to lightcast for skill extraction. 

Note: Unit tests for data transformation does sufficient job but please be sure to run this on local too.

### Testing Instructions for Contentful data transformation
1. Set the following variables in `private.py` file:
```
CONTENTFUL_SPACE_ID = <space_id>
CONTENTFUL_CONTENT_DELIVERY_API_KEY = <api_key>
BOOTCAMP_CONTENTFUL_CONTENT_TYPE = 'bootCampPage'
```
2. Run python shell in discovery shell: `make discovery-shell` `./manage.py shell`
3. Run the following code in shell to read the data from contentful:
```
from course_discovery.apps.course_metadata.utils import transform_bootcamp_contentful_data
transform_bootcamp_contentful_data()
```

### Testing Instructions for Algolia Reindexing on local
1. Set the following variables in `private.py` file (`MyAppID` is the id for EdX_Staging application and `MyApiKey` is the admin key of algolia):
```
ALGOLIA = {
    'APPLICATION_ID': 'MyAppID',
    'API_KEY': 'MyApiKey'
}
```
2. Apply this patch to your current github branch: [algolia_indexing_test_data.patch.gz](https://github.com/openedx/course-discovery/files/10205688/algolia_indexing_test_data.patch.gz) using this command `gunzip < /path/to/algolia_indexing_test_data.patch.gz | git apply -`
3. Edit the uuid on L126 with your local devstack `Demo_Course` uuid
4. Run discovery shell: `make discovery-shell`
5. Run the following: `./manage.py algolia_reindex`
6. See the new fields being indexed here for your demo course: https://www.algolia.com/apps/testing2MLYZ9WWU0/explorer/browse/product?searchMode=search

### Pre-merge:
- [x] Make an edx-internal [PR](https://github.com/edx/edx-internal/pull/7514) to add `CONTENTFUL_SPACE_ID` and `CONTENTFUL_CONTENT_DELIVERY_API_KEY` settings for stage and prod.

### Post-merge:
- [ ] Run Algolia-reindexing on staging and verify bootcamps data properly updated on EdX-Staging
